azure-ts-dynamicresource/cdnCustomDomain.ts

```diff
@@ -0,0 +1,244 @@
+import * as pulumi from "@pulumi/pulumi";
+import * as azure from "@pulumi/azure";
+
+import * as msRestAzure from "@azure/ms-rest-nodeauth";
+import * as cdnManagement from "@azure/arm-cdn";
+import { ServiceClientCredentials } from "@azure/ms-rest-js";
+
+export interface CustomDomainOptions {
+  resourceGroupName: string;
+  profileName: string;
+  endpointName: string;
+  customDomainHostName: string;
+  httpsEnabled: boolean;
+}
+
+export interface CustomDomainOutputs extends cdnManagement.CdnManagementModels.CustomDomainsCreateResponse {
```

Comment:
Is this an internal structure to bring the state between multiple runs or will a user interact with it? If the former, why is it exported? If the latter, its shape seems to be different from what a typical component would expose as outputs? Is there guidance for dynamic providers in this regards?

Reply:
> Is this an internal structure to bring the state between multiple runs or will a user interact with it? If the former, why is it exported?

I exported it because I am under the assumption that the type of the outputs objects could be used by the caller to extract specific properties. But I am not sure if it is necessary, and if there is a more "accepted" way for dynamic providers to declare an output type or if one is even needed. At the very least I could un-export this type.

> Is there guidance for dynamic providers in this regards?

I am going to speak to someone here at Pulumi about dynamic provider outputs too and have an answer for this.